File: client/src/main/kotlin/com/seleneworlds/client/game/GameApi.kt
package com.seleneworlds.client.game

import com.seleneworlds.client.window.WindowManager

class GameApi(
    private val windowManager: WindowManager
) {
    fun setWindowAspectRatio(width: Int, height: Int) {
        windowManager.setAspectRatio(width, height)
    }

    fun clearWindowAspectRatio() {
        windowManager.clearAspectRatio()
    }
}

File: client-web/src/api/GameApi.ts
export type ClientGamePreTickListener = () => void;

export interface GameApi {
  addPreTickListener: (listener: ClientGamePreTickListener) => () => void;
  setWindowAspectRatio: (width: number, height: number) => void;
  clearWindowAspectRatio: () => void;
  setOffscreenRendering: (width: number, height: number) => void;
  setNativeRendering: () => void;
}
